wandb/sdk/data_types/image.py

```diff
@@ -508,29 +560,30 @@ def guess_mode(
             )
 
     @classmethod
-    def to_uint8(cls, data: "np.ndarray") -> "np.ndarray":
-        """Convert image data to uint8.
-
-        Convert floating point image on the range [0,1] and integer images on the range
-        [0,255] to uint8, clipping if necessary.
-        """
+    def normalize(cls, data: "np.ndarray") -> "np.ndarray":
+        """Normalizes and converts image pixel values to uint8 in the range [0, 255]."""
         np = util.get_module(
             "numpy",
             required="wandb.Image requires numpy if not supplying PIL Images: pip install numpy",
         )
 
-        # I think it's better to check the image range vs the data type, since many
-        # image libraries will return floats between 0 and 255
-
-        # some images have range -1...1 or 0-1
-        dmin = np.min(data)
-        if dmin < 0:
+        # if an image has negative values, set all values to be in the range [0, 1]
+        # This can lead to inconsistent behavior when an image has only a single negative value
+        if np.min(data) < 0 and np.ptp(data) != 0:
             data = (data - np.min(data)) / np.ptp(data)
+
         if np.max(data) <= 1.0:
-            data = (data * 255).astype(np.int32)
+            data = (255 * data).astype(np.int32)
 
-        # assert issubclass(data.dtype.type, np.integer), 'Illegal image format.'
-        return data.clip(0, 255).astype(np.uint8)
+        return data.clip(0, 255)
+
+    @classmethod
+    def convert_to_uint8(cls, data: "np.ndarray") -> "np.ndarray":
```

Comment:
Likewise, should be prefixed with an underscore

Also, doesn't need to be a classmethod